doc/versioning.md

```diff
@@ -0,0 +1,61 @@
+# Versioning
+
+For better managing and releasing code, here is a summary about the versioning story of ElasticDL. Some previous discussions can be found [here](#1121).
+
+## Branches
+
+There are three different type branches:
+- **develop** branch: the main branch where the source code always reflects a state with the latest delivered development changes for the next release.
+- **feature** branches: branch off from the `develop` branch and merge back into `develop`. `Feature` branches are used to develop new features, and created/maintained by developers themselves. `Feature` branches will be deleted after code changes get merge info `develop` branch.
+- **release** branches: branch off from the `develop` branch. When the `develop` branch reaches the desired state of a new release, cut off a new release branch, and start to publish new versions there. Release branches will be kept.
+
+For version numbers, we follow the [Semantic Versioning 2.0.0](https://semver.org/ with style MAJOR.MINOR.PATCH (e.g. 0.2.1). For each MAJOR.MINOR, we keep a corresponding `release` branch (named `release-x.y`), and each following patch version x.y.z is released from it.
```

Comment:
Would it be better to be more explicit MAJOR.MINOR.0 instead of MAJOR.MINOR?

Reply:
This should be MAJOR.MINOR, for example, if we say 1.2 here, the release version 1.2.0, 1.2.1, 1.2.2... will be released from this branch. So if we use 1.2.0, it will make confusion..

Reply:
Sounds good